proposals/0279-screen-manager-subscribe-buttons.md

```diff
@@ -34,16 +33,23 @@ public interface OnButtonListener {
     void onError(String info);
 }
 
-public void subscribeButton(ButtonName buttonName, OnButtonListener listener);
-public void unsubscribeButtonListener(OnButtonListener listener);
+public void addButtonListener(ButtonName buttonName, OnButtonListener listener);
+public void removeButtonListener(ButtonName buttonName, OnButtonListener listener);
 ```
 
 ### JavaScript Suite
 The JavaScript Suite APIs would be set up in a similar way to the iOS and Java Suite APIs above. Any necessary changes are at the discretion of the Project Maintainer. However larger changes that would impact the iOS or Java Suite code above (such as adding or removing a method) will require proposal revisions.
 
 ### Implementation Notes
 * There will still need to be storage for the blocks and observers, and this will be handled by a new sub-manager.
-* When the first subscription is added for a button, the `SubscribeButton` RPC should be sent, when the last subscription is removed, the `UnsubscribeButton` RPC should be sent.
+* When the first subscription is added for a button, the `SubscribeButton` RPC will be sent, when the last subscription is removed, the `UnsubscribeButton` RPC will be sent.
+* The sub-manager will not attempt to check for correct permissions and will just send the request. Any errors returned by core will be passed to the observer via the `error`  parameter on the `handler`/`listener`. 
+*  The iOS selector can have the following arguments:
+    1. A selector with no parameters. The observer will be notified only when a short button press occurs.
```

Comment:
Thinking about this more. I think it should notify for a short or long press, but there's no way to differentiate between them (so the dev can only do one thing for whichever occurs).